staubli_val3_driver/val3/ros_server/decTrajPtFull.pgx

```diff
@@ -32,54 +32,65 @@
 
   // robotId
   l_nRetVal=fromBinary(rosTrajPtFMsg.body.nData,4,"-4l",rosTrajPtFMsg.jointTrajPtFull.nRobotId)
-  if (l_nRetVal!=1)
+  if(l_nRetVal!=1)
     x_bFlag=false
     return
   endIf
 
   // sequence
   l_nRetVal=fromBinary(rosTrajPtFMsg.body.nData,4,"-4l",rosTrajPtFMsg.jointTrajPtFull.nSeq)
-  if (l_nRetVal!=1)
+  if(l_nRetVal!=1)
     x_bFlag=false
     return
   endIf
 
   // valid fields
   l_nRetVal=fromBinary(rosTrajPtFMsg.body.nData,4,"-4l",rosTrajPtFMsg.jointTrajPtFull.nValidFields)
-  if (l_nRetVal!=1)
+  if(l_nRetVal!=1)
     x_bFlag=false
     return
   endIf
 
   // time
   l_nRetVal=fromBinary(rosTrajPtFMsg.body.nData,4,"4.0l",rosTrajPtFMsg.jointTrajPtFull.nTime)
-  if (l_nRetVal!=1)
+  if(l_nRetVal!=1)
     x_bFlag=false
     return
   endIf
 
   // positions[10]
   for l_nIndex=0 to 9 step 1
     l_nRetVal=fromBinary(rosTrajPtFMsg.body.nData[(l_nIndex*4)+16],4,"4.0l",rosTrajPtFMsg.jointTrajPtFull.nPositions[l_nIndex])
-    if (l_nRetVal!=1)
+    if(l_nRetVal!=1)
       x_bFlag=false
       return
     endIf
   endFor
 
   // velocities[10]
   for l_nIndex=0 to 9 step 1
+    // Velocity buffer to smooth out last traj point
+    nVel[1] = nVel[0]
+    nVel[0] = rosTrajPtMsg.jointTrajPt.nVelocity
```

Comment:
Watch out here, this looks like a typo due to copy&paste. The variable in the `decTrajPtFull` is not called `rosTrajPtMsg` but `rosTrajPtFMsg`.

Reply:
and consequently, `rosTrajPtMsg.jointTrajPt` does not exist and should instead be `rosTrajPtFMsg.jointTrajPtFull`

Reply:
This bit needs to be rewritten, the copy&paste does not work, `nVelocity` is also incorrect.

Reply:
I see, my mistake. @gavanderhoorn do you know which in which cases `decTrajPtFull` is used? in my testing with MoveIt, I never saw it used.

Reply:
The IRC doesn't use `JOINT_TRAJ_PT_FULL` right now.

It could, but it doesn't.

I would recommend to keep this code around though.

Reply:
Actually it's a bit of a liability, it's untestable and it makes adding any other improvements harder. If given the choice I would prefer removing code like this, and have a more streamlined code base.

Reply:
My comment was specifically about the decoders for `JOINT_TRAJ_PT_FULL`, not about the additions that @marshallpowell97 contributes in this PR.

The decoder is fine, and I already commented on the fact that I don't like the 'interpolation' to be done in the decoder. So that should be placed somewhere else, regardless of whether the decoder is used or not.

Reply:
Yes, I understand. My comment is a general remark about the problems associated with code that is dead code for all practical purposes. As you know, I recently worked on extending this driver as well, and I went through similar issues as @marshallpowell97: I kept copying&pasting code from the "non-full" implementation into this part and was unable to test because nothing really exercises this code path. The effort is not insignificant and it represents a barrier for contribution. Even in this pull request, the suggestion to refactor this to avoid code duplication means extra work that is not going to be used. It might even be buggy and end up in production because no one can really test.

So, as a general opinion, I would strongly favor an approach to promote smaller code base and actively remove functionality like this, that is unreachable, or untestable, or dead, or half-implemented.

Reply:
> Even in this pull request, the suggestion to refactor this to avoid code duplication means extra work that is not going to be used. It might even be buggy and end up in production because no one can really test.

I'm not sure I understand. @marshallpowell97 should not have added anything in these two programs.

How would that have led to duplication of effort?

Reply:
> So, as a general opinion, I would strongly favor an approach to promote smaller code base and actively remove functionality like this, that is unreachable, or untestable, or dead, or half-implemented.

I'm not sure this is half-implemented: a proper implementation of a `simple_message` server library should just support a nr of different message types. `JOIN_TRAJ_PT_FULL` is one of those. It was added with the express intent of improving the server part of the driver (ie: this VAL 3 application) to start using it.

Reply:
In any case, I believe this is somewhat out-of-scope for this PR.

To move this forward:

 - @marshallpowell97: please revert all changes to the `JOINT_TRAJ_PT_FULL` decoder program, and
 - please move/implement the averaging outside of the decoder for the regular joint traj pt

We can then proceed with the review and tests of this PR.

As @gonzalocasas' point is valid, we will track this (ie: dead code) in a separate issue and then deal with it at a later time. My suggestion would actually be to add some tests, which exercise relevant / reachable parts of the code sufficiently to have trust in them.

That would be necessary anyway, as a large part of the code in this repository is a library. By definition there will be code in it that is not always used, but should still be there to offer a sufficient set of reusable functionality.

Would that be acceptable @gonzalocasas?

Reply:
Perfect! Thanks!